src/new-banlist.ts

```diff
@@ -0,0 +1,174 @@
+});
+let addCards = new Array<string>();
+if (options.add_cards) {
+  addCards = options.add_cards.split(',');
+}
+const addPrintings = new Array<string>();
+addCards.forEach(c => {
+  addPrintings.push(cardsToPrintings[c]);
+});
+let removeCards = new Array<string>();
+if (options.remove_cards) {
+  removeCards = options.remove_cards.split(',');
+}
+const removePrintings = new Array<string>();
+removeCards.forEach(c => {
+  removePrintings.push(...(cardsToPrintings.get(c) || []));
+});
+
+
+let srcMwl: any = null;
+const mwls = getMwlJson();
+mwls.forEach(m => {
+  if (m.code == options.source_code) {
+    srcMwl = JSON.parse(JSON.stringify(m));
+  }
+});
+
+console.log(`Found mwl ${JSON.stringify(srcMwl)}`);
+
+srcMwl.code = options.new_code;
+srcMwl.name = options.name;
+srcMwl.date_start = options.effective_date;
+addCards.forEach(c => {
+  cardsToPrintings.get(c)?.forEach(p => {
+    srcMwl.cards[p] = { "deck_limit": 0 };
+  });
+});
+
+// Helpers to deal with the fact that the MWL cards entry is a map-like object.
+function convertObjectToMap<T>(obj: { [key: number]: T }): Map<string, T> {
+  const map = new Map<string, T>();
+  for (const key in obj) {
+      if (Object.prototype.hasOwnProperty.call(obj, key)) {
+          map.set(key, obj[key]);
+      }
+  }
+  return map;
+}
+
+function filterMapKeys<T, U>(sourceMap: Map<T, U>, keysToFilterOut: T[]): Map<T, U> {
+  const filteredMap = new Map<T, U>();
+  for (const [key, value] of sourceMap) {
+      if (!keysToFilterOut.includes(key)) {
+          filteredMap.set(key, value);
+      }
+  }
+  return filteredMap;
+}
+
+function convertMapToObject<T>(map: Map<string, T>): { [key: number]: T } {
+  const obj: { [key: number]: T } = {};
+  for (const [key, value] of map.entries()) {
+    obj[key] = value;
+  }
+  return obj;
+}
+
+srcMwl.cards = convertMapToObject(filterMapKeys(convertObjectToMap(srcMwl.cards), removePrintings));
+
+mwls.push(srcMwl);
+
+formatJSON(mwls).then((content) => {
+    fs.writeFile(
+        resolve("mwl.json"),
+        content,
+        function (err) {
+            if (err) {
+                console.log(`Could not write updated mwls.json: ${err}`);
+            }
+        },
+    );
+});
+
+// Handle the V2 restriction format.
+const restrictions = getRestrictionsV2Json();
+
+const standardRestrictions = restrictions['standard'];
+
+let srcRestriction: any = null;
+
+standardRestrictions.forEach((v) => {
+  console.log(`${v.id}`);
+  if (textToId(options.source_code) == v.id) {
+    srcRestriction = v;
+  }
+});
+
+srcRestriction.date_start = options.effective_date;
+srcRestriction.name = options.name;
+srcRestriction.id = textToId(options.new_code);
+srcRestriction.banned.push(...addCards);
+srcRestriction.banned = srcRestriction.banned.filter((c: string) => !removeCards.includes(c));
+// TODO(plural): Remove this when prettier's jsonRecursiveSort is fixed.
+srcRestriction.banned = srcRestriction.banned.sort();
```

Comment:
`Array.prototype.sort` does the sorting in-place, so no need to re-assign it here. [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)